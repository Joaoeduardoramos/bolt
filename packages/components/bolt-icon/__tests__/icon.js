const { render } = require('@bolt/twig-renderer');

describe('<bolt-icon> Component', () => {
  test('basic usage', async () => {
    const results = await render('@bolt-components-icon/icon.twig', {
      name: 'add-open',
      background: 'square',
      size: 'medium',
      color: 'teal',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic usage with attributes', async () => {
    const results = await render('@bolt-components-icon/icon.twig', {
      attributes: {
        style: [
          '--bolt-theme-icon-background: #e64b18;',
          '--bolt-theme-icon: #FFF;',
          '--bolt-theme-icon-background-opacity: 1;',
        ],
      },
      name: 'add-open',
      background: 'square',
      size: 'medium',
      color: 'teal',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
