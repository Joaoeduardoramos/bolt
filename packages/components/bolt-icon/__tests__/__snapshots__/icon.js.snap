// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-icon> Component basic usage with attributes 1`] = `

<bolt-icon name="add-open"
           background="square"
           size="medium"
           color="teal"
>
</bolt-icon>

`;
